community_supplied/Linecard/netconf-fpc-cpu-stats.rule

```diff
@@ -0,0 +1,261 @@
+                        field-name "$memory-buffer-utilization";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory buffer utilization";
+            }
+            field dt-cpu-total {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$cpu-total";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for CPU utilization";
+            }
+            field dt-heap-util {
+                formula {
+                    dynamic-threshold {
+                        algorithm k-means;      
+                        learning-period 7d;
+                        pattern-periodicity 1d;
+                        field-name "$memory-heap-utilization";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory heap utilization";
+            }
+            field slot {
+                sensor fpc-sensor {
+                    where "slot =~ /{{slot-name}}/";
+                    path slot;
+                }
+                type string;
+                description "fpc slot number";
+            }
+            field threshold {
+                constant {
+                    value "{{static-threshold}}";
+                }
+                type integer;
+                description "Threshold value for memory & CPU utilization values";
+            }
+			/*
+            * Anomaly detection logic for buffer memory utilization.
+            */
+            trigger buffer-memory-utilization {
+                frequency 300s;
+				/*
+                * Sets color to red and sends out an anomaly notification when
+                * memory buffer utilization count increases more than threshold static ($threshold) value
+                * 
+                */
+                term is-buffer-memory-utilization-is-abnormal {
+                    when {
+                        greater-than "$memory-buffer-utilization" "$threshold";
+                    }
+                    then {                      
+                        status {
+                            color red;
+                            message "Buffer memory utilization for fpc slot=($slot) is crossed static threshold value ($threshold), Value=($memory-buffer-utilization)";
+                        }
+                    }
+                }
+				/*
+                * Sets color to yellow and sends out an anomaly notification when
+                * memory buffer utilization count increases more than dynamic threshold ($threshold) value
+                * 
+                */
+                term is-buffer-memory-utilization-is-minimal {

```

Comment:
The term name is misleading -since it says the buffer memory utilization is minimal, which is a good thing. What we intend to say is that has-buffer-memory-utilization-breached-dynamic-threshold? We should fix this for all dynamic-threshold based yellow alarms..


Please fix all instances of dt checks to something along

Reply:
Modified.